granulate_utils/linux/cgroups/base_cgroup.py

```diff
@@ -0,0 +1,96 @@
+#
+# Copyright (c) Granulate. All rights reserved.
+# Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
+#
+
+from enum import Enum
+from pathlib import Path
+from typing import List
+
+from granulate_utils.linux.cgroups.exceptions import MissingCgroup, MissingController, SkippedCgroup, UnsupportedCgroup
+
+PID_CGROUPS = Path("/proc/self/cgroup")
+CGROUP_PARENT_PATH = Path("/sys/fs/cgroup")
+
+
+class CGROUPS(Enum):
+    memory = "memory"
+    cpu = "cpu,cpuacct"
+
+
+class CgroupVerifications:
+    @staticmethod
+    def is_cgroup_v1() -> None:
+        # "/proc/$PID/cgroup" lists a process's cgroup membership.  If legacy
+        # cgroup is in use in the system, this file may contain multiple lines,
+        # one for each hierarchy.
+        # The entry for cgroup v2 is always in the format "0::$PATH"::
+        if len(PID_CGROUPS.read_text().split("\n")) == 2:
+            raise UnsupportedCgroup("version 2")
+
+    @staticmethod
+    def is_supported_cgroup(cgroup_type: str) -> None:
+        if cgroup_type not in CGROUPS.__members__:
+            raise UnsupportedCgroup(cgroup_type)
+
+    @staticmethod
+    def is_ignored_cgroup(cgroup: str, custom_cgroup: str) -> None:
```

Comment:
This is actualy the opposite - unlike `is_supported_cgroup` which asserts the subsystem exists, this one asserts that it is NOT ignored